### ISSUE TITLE
Golint is deprecated

### ISSUE THREAD

**Issue by antichris** [gojp/goreportcard#378]
> **NOTE:** Golint is [deprecated and frozen](https://github.com/golang/go/issues/38968). There's no drop-in replacement for it, but tools such as [Staticcheck](https://staticcheck.io/) and `go vet` should be used instead.
> — https://github.com/golang/lint#readme

This was already brought up in #349, but the issue was closed by the bot without anyone actually addressing it (which is probably yet another issue: when nobody gives a shit about an issue for more than 60 days, it just magically "poof"s closed).

**Comment by antichris** [comment]
Also CC: @cristaloleg, @TheDiveO, @BrenekH, @Denchick, @georgettica, @obalunenko, @antoineco, @benjaminbartels, @slyang-git, and @patrickhoefler, who thumbs-up'ed both previous issues and, I suspect, would appreciate this issue being finally addressed for real.

**Comment by cristaloleg** [comment]
Yes, `golint` is frozen but not deprecated, it still works and suggests what is better to fix. However, after Go 1.18 release `golint` (probably?) will not work with generics because it knows nothing about them.

AFAIR https://github.com/mgechev/revive is a drop-in replacement for `golint` and probably that's the tool that we should use.

**Comment by antichris** [comment]
> Yes, `golint` is frozen but not deprecated, it still works and suggests what is better to fix

If by "not deprecated" you mean "it can still be downloaded, installed and kind of more or less works", I would hesitantly agree.

But the fact is that the repo is archived, the package is officially unmaintained and frozen, and the tens of outstanding issues it had were all closed unresolved when the Go team agreed on and enacted the decision to, literally, ***deprecate*** (https://github.com/golang/go/issues/38968#issuecomment-638331245) it. The tool is now abandoned and the recommendation stands against using it. Precisely as I understand "deprecated".

Revive is good, though. :+1:

**Comment by peczenyj** [comment]
Hello

the latest version of golangci-lint (v1.45.0) add some support to go1.18 via one flag that disable most of the problematic linters that does not work properly with generics.

I try to use goreportcard with one project using generics and the result was a big mess. Perhaps it is time to move forward?

https://github.com/golangci/golangci-lint/releases/tag/v1.45.0

**Comment by antichris** [comment]
@shawnps 
> (...) the golint check has been removed.
> — https://github.com/gojp/goreportcard/issues/381#issuecomment-1074672143

Is that the final resolution for this issue or are there other plans (e.g. [revive], as https://github.com/gojp/goreportcard/issues/378#issuecomment-1039031222 suggested, or something else)?

[revive]: https://github.com/mgechev/revive

**Comment by thediveo** [comment]
As revive stresses that out of the box it acts as a faster drop-in replacement for golint, please drop it finally into goreportcard! From reading and rereading this issue and revive's documentation that shouldn't be rocket science and get this issue quickly and reliably plugged?!
